lib/utils/config.ts

```diff
@@ -134,3 +134,34 @@ export function generateDeviceConfig(
 			return config;
 		});
 }
+
+/**
+ * Chech whether the `--dev` option of commands related to OS configuration
+ * such as `os configure` and `config generate` is compatible with a given
+ * balenaOS version, and print a warning regarding the consequences of using
+ * that option.
+ */
+export async function validateDevOptionAndWarn(
+	dev?: boolean,
+	version?: string,
+	logger?: import('./logger'),
+) {
+	if (!dev) {
+		return;
+	}
+	if (version && (version.includes('.prod') || version.includes('+prod'))) {
```

Comment:
No `\b`? :disappointed: 
```suggestion
	if (version && /\bprod\b/test(version)) {
```

Reply:
Done 👍